sound/soc/sof/pm.c

```diff
@@ -174,6 +177,9 @@ static int sof_suspend(struct device *dev, bool runtime_suspend)
 	if (!sof_ops(sdev)->suspend)
 		return 0;
 
+	if (sdev->fw_state == SOF_FW_BOOT_NOT_STARTED)
+		return 0;
+
```

Comment:
do you actually need this change? If you keep the code intact, the flow will use the next test to go to suspend: and you test there.

You can argue that it saves one test I guess? 
On the other hand, you have an addittional jump to suspend: below, so wondering if we break @ranj063 's work with this?

Reply:
@plbossart Btw, I still think #1746 is better, please see my comments there :) 

But yeah, I think this is valid point. I had two checks for NOT_STARTED in first version I pushed, and I removed only one.. but yeah, no reason to change the original code in suspend.